components/tenant-fetcher/internal/tenant/converter.go

```diff
@@ -14,21 +16,36 @@ func NewConverter() *converter {
 func (c *converter) ToEntity(in model.TenantModel) tenant.Entity {
 	return tenant.Entity{
 		ID:             in.ID,
-		Name:           in.TenantId,
+		Name:           in.Name,
 		ExternalTenant: in.TenantId,
-		ProviderName:   in.TenantProvider,
+		Parent:         newNullString(in.ParentInternalId),
+		Type:           in.Type,
+		ProviderName:   in.Provider,
 		Status:         in.Status,
 	}
 }
 
-func (c *converter) FromEntity(in *tenant.Entity) *model.TenantModel {
-	if in == nil {
-		return nil
+func (c *converter) FromEntity(in tenant.Entity) model.TenantModel {
+	return model.TenantModel{
+		ID:               in.ID,
+		Name:             in.Name,
+		TenantId:         in.ExternalTenant,
+		ParentInternalId: in.Parent.String,
+		Type:             in.Type,
+		Provider:         in.ProviderName,
+		Status:           in.Status,
 	}
-	return &model.TenantModel{
-		ID:             in.ID,
-		TenantId:       in.ExternalTenant,
-		TenantProvider: in.ProviderName,
-		Status:         in.Status,
+}
+
+func newNullString(s string) sql.NullString {
```

Comment:
Code duplication - this function was already available in components/director/internal/domain/tenant/converter.go.

Probably we may consider extracting it and reusing it in common place for both packages like: github.com/kyma-incubator/compass/components/director/pkg/tenant package?

Reply:
Will be addressed in next change